cooking-app/src/App.js

```diff
@@ -0,0 +1,56 @@
+import { BrowserRouter } from "react-router-dom";
+import { GlobalStyle, Wrapper } from "./components/styles/Global.styled";
+import { Header } from "./components/Header/Header";
+import { Content } from "./components/Content/Content";
+import { Footer } from "./components/Footer/Footer";
+import { AddRecipe } from "./components/Content/AddRecipes/AddRecipe";
+import { useEffect, useState } from "react";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth, db, usersCollection } from "./api/firebase";
+import { createContext } from "react";
+import { doc, getDoc } from "firebase/firestore";
+import { Loader } from "./utils/Loader";
+
+export const UserDataContext = createContext("ulabula");
+
+function App() {
+  const [userData, setUserData] = useState(null);
+  const [isLoggedIn, setIsLoggedIn] = useState(null);
+
+  useEffect(() => {
+    onAuthStateChanged(auth, (user) => {
+      if (user) {
+        setIsLoggedIn(true);
+        console.log(user);
+        getDoc(doc(db, "users", user.uid)).then((querySnaphot) => {
+          const tempData = querySnaphot.data();
+          setUserData({
+            firstName: tempData.firstName,
+            lastName: tempData.lastName,
+            email: tempData.email,
+            role: tempData.role,
+            uid: tempData.uid,
+          });
+        });
+      } else {
+        setIsLoggedIn(false);
+        setUserData(null);
+      }
+    });
+  }, []);
+
+  return (
+    <BrowserRouter>
+      <GlobalStyle />
+      <UserDataContext.Provider value={userData}>
+        <Wrapper>
+          <Header isLoggedIn={isLoggedIn} />
+          {isLoggedIn === null ? <Loader /> : <Content isLoggedIn={isLoggedIn} />}
+          <Footer />
```

Comment:
Czy możemy zastąpić zastąpić "isLoggedIn === null, "!isLoggedIn", jak myślicie?

Reply:
obawiam się, że nie

wartość null jest specyficzna i pożądana, w momencie uruchomienia strony aplikacja potrzebuje trochę czasu żeby ustalić czy użytkownik jest zalogowany czy nie, na ten czas isLoggedIn === null i wyświetlany jest <Loader />

jeśli żaden użytkownik nie jeste zalogowany to isLoggedIn przyjmuje false

isLoggedIn równy false sprawiłby, że <Loader /> nigdy nie zniknie jeśli użytkownik nie jest zalogowany

null !== false